multi scale sampler and dataset

Sibo2rr opened the pull request.

RainFrost1 committed:
add shitu deploy doc

Commit:
add en doc

Commit:
add en doc

RainFrost1 committed:
add English doc for ImageClassifcation,ImageRecognition, VectorSearch, MainBodyDetection

weisy11 committed:
init fuse gallery

weisy11 committed:
move file to ppcls

weisy11 committed:
update functions

weisy11 committed:
dbg

weisy11 committed:
add logger

Commit:
rename en doc;modify some errors

TingquanGao committed:
feat: support PVTV2

TingquanGao committed:
feat: support PVTV2

TingquanGao committed:
docs: update en docs

weisy11 committed:
update gallery2fc

weisy11 committed:
add feature extract

RainFrost1 committed:
add en docs for slim and cpp_shitu

TingquanGao committed:
fix: fix key error in distillation

Sibo2rr committed:
build html of read the doc

TingquanGao committed:
fix: fix -1 in dims and ignore swapdim in static

TingquanGao committed:
rename

TingquanGao committed:
fix: fix -1 in dims

Sibo2rr committed:
multi scale sampler and dataset

Sibo2rr committed:
multi scale sampler and dataset

Sibo2rr committed:
add config file

Sibo2rr committed:
add config file

TingquanGao committed:
docs: update benchmark curve

TingquanGao committed:
docs: resize pic

cuicheng01 committed:
Merge pull request #1570 from TingquanGao/dev/update_benchmark

docs: update benchmark curve

RainFrost1 committed:
fix levit384 inference size bug

TingquanGao committed:
feat: add doc and tipc config of pvtv2

Sibo2rr committed:
read the doc

TingquanGao committed:
fix: fix lr

TingquanGao commented on ppcls/data/dataloader/multi_scale_sampler.py:
```diff
@@ -0,0 +1,105 @@
+
+class MultiScaleSamplerDDP(Sampler):
+    def __init__(self, data_source, scales, first_bs, g):
+        print(scales)
```
如果是需要给用户打印出MultScale的信息，使用 `logger`
```python
from ppcls.utils import logger
```

TingquanGao commented on ppcls/data/dataloader/multi_scale_sampler.py:
```diff
@@ -0,0 +1,105 @@
+        base_batch_size = first_bs
+        
+        # Get the GPU and node related information
+        num_replicas  =dist.get_world_size()
```
pre-commit 一下

TingquanGao commented on ppcls/data/dataloader/multi_scale_sampler.py:
```diff
@@ -0,0 +1,105 @@
+            # ImageNet models down-sample images by a factor of 32.
+            # Ensure that width and height dimensions are multiples are multiple of 32.
+            width_dims = [int((w // 32) * 32) for w in width_dims]
+            height_dims = [int((h // 32) * 32) for h in height_dims]
```
这里为什么选32

TingquanGao commented on ppcls/data/dataloader/multi_scale_sampler.py:
```diff
@@ -0,0 +1,105 @@
+        if self.shuffle:
+            random.seed(self.epoch)
+            random.shuffle(self.img_indices)
+            random.shuffle(self.img_batch_pairs)
```
问一下水龙/崔程，该处的随机变量是否需要能通过配置固定种子。

TingquanGao commented on ppcls/data/dataloader/multi_scale_sampler.py:
```diff
@@ -0,0 +1,105 @@
+            batch_ids = indices_rank_i[start_index:end_index]
+            n_batch_samples = len(batch_ids)
+            if n_batch_samples != curr_bsz:
+                    batch_ids += indices_rank_i[:(curr_bsz - n_batch_samples)]
```
缩进似乎有问题，过一下pre-commit

TingquanGao commented on ppcls/data/dataloader/multi_scale_sampler.py:
```diff
@@ -0,0 +1,105 @@
+
+            if len(batch_ids) > 0:
+                    batch = [(curr_h, curr_w, b_id) for b_id in batch_ids]
+                    yield batch
```
缩进似乎有问题

TingquanGao commented on ppcls/data/dataloader/multi_scale_dataset.py:
```diff
@@ -0,0 +1,119 @@
+        self._cls_path = cls_label_path
+        self.transform_ops = transform_ops
+        # if transform_ops:
+        #     self._transform_ops = create_operators(transform_ops)
```
是否需要删掉？

TingquanGao commented on ppcls/data/dataloader/multi_scale_dataset.py:
```diff
@@ -0,0 +1,119 @@
+                self.labels.append(np.int64(l[1]))
+                assert os.path.exists(self.images[-1])
+
+
```
类内方法间空一行

TingquanGao commented on ppcls/data/dataloader/multi_scale_dataset.py:
```diff
@@ -0,0 +1,119 @@
+        if self.transform_ops:
+            for i in range(len(self.transform_ops)):
+                op = self.transform_ops[i]
+                if 'RandCropImage' in op:
```
除了`RandCropImage`，还有`ResizeImage`等op是否需要一同考虑进来。

TingquanGao commented on ppcls/data/dataloader/multi_scale_dataset.py:
```diff
@@ -0,0 +1,119 @@
+            for i in range(len(self.transform_ops)):
+                op = self.transform_ops[i]
+                if 'RandCropImage' in op:
+                    warnings.warn("Multi scale dataset will crop image according to the multi scale resolution")
```
使用`logger`

TingquanGao commented on ppcls/data/dataloader/multi_scale_dataset.py:
```diff
@@ -0,0 +1,119 @@
+                op = self.transform_ops[i]
+                if 'RandCropImage' in op:
+                    warnings.warn("Multi scale dataset will crop image according to the multi scale resolution")
+                    self.transform_ops[i]['RandCropImage'] = {'size': img_width}
```
如果 `img_width` 和 `img_height` 不相等，该处参数是否有问题？

TingquanGao commented on ppcls/data/dataloader/multi_scale_dataset.py:
```diff
@@ -0,0 +1,119 @@
+                    self.transform_ops[i]['RandCropImage'] = {'size': img_width}
+                    has_crop = True
+        if has_crop == False:
+            raise RuntimeError("Multi scale dateset requests RandCropImage")
```
先用`logger`打印error msg，再抛异常

TingquanGao commented on ppcls/data/dataloader/multi_scale_dataset.py:
```diff
@@ -0,0 +1,119 @@
+from ppcls.utils import logger
+
+
+def create_operators(params):
```
该函数代码是否可以复用之前的？

TingquanGao commented on ppcls/configs/multi_scale/MobileNetV1_multi_scale.yaml:
```diff
@@ -0,0 +1,133 @@
+
+    sampler:
+      name: MultiScaleSamplerDDP
+      scales: [160, 192, 256, 288, 320]
```
可以加上一行：
```yaml
# support to specify width and height respectively:
# scales: [(160,160), (192,192), (256,256) (288,288) (320,320)]
```

Commit:
update paddle serving on dev branch

Sibo2rr committed:
modify according to review

Sibo2rr committed:
resolve conflict

Sibo2rr committed:
fix images error

Sibo2rr committed:
add title

Sibo2rr committed:
fix error

TingquanGao committed:
fix: unify the pattern of layer's name

cuicheng01 committed:
update en docs

TingquanGao committed:
fix: fix result returned by _find_layers_handle

TingquanGao committed:
fix: fix result returned by stop_after

zhangbo9674 committed:
Accelerate dynamic graph amp training

weisy11 committed:
add 2 new docs

TingquanGao committed:
fix: fix comments

RainFrost1 committed:
fix Rednet101 and Swin_large12_384 bs bug for tipc

cuicheng01 committed:
update dali & course md

Commit:
add classification and recogniton dataset md

cuicheng01 committed:
add some new configs

weisy11 committed:
update format and add function_intro_en

weisy11 committed:
update format

cuicheng01 commented on ppcls/configs/multi_scale/MobileNetV1_multi_scale.yaml:
```diff
@@ -0,0 +1,137 @@
+    sampler:
+      name: MultiScaleSamplerDDP
+      scales: [160, 192, 256, 288, 320]
+      first_bs: 64
```
1.这里需要简单说明下first_bs, down_sample是啥，另外，我们讨论down_sample是不是换了divided factor了。
2.示例内容的base尺寸选用224吧。

cuicheng01 commented on ppcls/data/__init__.py:
```diff
@@ -28,11 +28,13 @@
 from ppcls.data.dataloader.DistributedRandomIdentitySampler import DistributedRandomIdentitySampler
 from ppcls.data.dataloader.pk_sampler import PKSampler
 from ppcls.data.dataloader.mix_sampler import MixSampler
+from ppcls.data.dataloader.multi_scale_sampler import MultiScaleSamplerDDP
```
这里的DDP建议删掉，下同

cuicheng01 commented on ppcls/data/dataloader/multi_scale_sampler.py:
```diff
@@ -0,0 +1,116 @@
+from ppcls import data
+
+class MultiScaleSamplerDDP(Sampler):
+    def __init__(self, data_source, scales, first_bs, divided_factor=32, is_training = True, seed=None):
```
注意下格式

TingquanGao committed:
docs: add en docs

TingquanGao committed:
docs: rename

Sibo2rr committed:
add file

Sibo2rr committed:
modify title

zhangbo9674 committed:
refine optimizer init logice

TingquanGao committed:
docs: fix link

zhangbo9674 committed:
refine code

TingquanGao committed:
fix: fix problems commented in reviewing

Commit:
update en doc

Commit:
Update classification_dataset_en.md

Commit:
Update recognition_dataset_en.md

Commit:
Update paddle_serving_deploy_en.md

Commit:
Update paddle_serving_deploy_en.md

Intsigstephon committed:
Add files via upload

Commit:
Update feature_extraction_en.md

TingquanGao committed:
refactor: extract _parse_pattern_str() func

TingquanGao committed:
docs: fix problems commented in reviewing

TingquanGao committed:
docs: fix problems commented in reviewing

TingquanGao committed:
docs: Contents -> Catalogue

TingquanGao committed:
docs: Contents -> Catalogue

TingquanGao committed:
docs: remove .gitkeep

TingquanGao committed:
refactor: strengthen parse_pattern_str() func

TingquanGao committed:
refactor: rename replace_sub() func to ?

Commit:
update serving doc for CPU deployment

RainFrost1 committed:
update catlogue and link

RainFrost1 committed:
update catlogue and link for slim en doc

RainFrost1 committed:
update catlogue and link for cpp shitu en doc

Commit:
Update paddle_serving_deploy.md

weisy11 committed:
Fix static training speed (#1590)

* fix training speed

* update config setting method

TingquanGao committed:
refactor: simplify code

1. remove WrapLayer and wrap_theseus;
2. support call update_res() one more;
3. optim parse_pattern_str() to return list of layer parsed.

TingquanGao committed:
add license

TingquanGao committed:
fix: remove unnecessary register_hook() call & pre-commit

weisy11 committed:
dbg

weisy11 committed:
add export method

weisy11 committed:
refactor code

weisy11 committed:
move gallery layer into extractor

RainFrost1 committed:
fix tipc lite_train_whole_infer bug

weisy11 committed:
dbg

cuicheng01 committed:
update README & ResNet50_vd_10w_pretrained_url

TingquanGao committed:
fix: raise warning when setting batch_transform_ops and TopkAcc

Commit:
Update metric_learning_en.md

Commit:
Update classification_dataset_en.md

Commit:
Update recognition_dataset_en.md

Commit:
Update feature_extraction_en.md

Commit:
Update paddle_serving_deploy_en.md

TingquanGao committed:
fix: save latest model every epoch

Sibo2rr committed:
remove tricks in models, add it in the models_training

weisy11 committed:
dbg

weisy11 committed:
add license info

weisy11 committed:
dbg

TingquanGao committed:
fix: rename to upgrade_sublayer()

weisy11 committed:
content->catalogue; add links

weisy11 committed:
dbg

TingquanGao committed:
fix: replace reg exp with layer name

TingquanGao committed:
docs: update wechat qr code

Commit:
upload cplusplus serving

Commit:
update some errors

Commit:
Update paddle_serving_deploy.md

Commit:
update cpp serving client

Commit:
update binary model config

Commit:
remove comments

Commit:
add newline flag

Commit:
format config

Commit:
add Enter flag

Commit:
add Enter flag1

Commit:
update quick_start.md

Commit:
Update quick_start_recognition.md

TingquanGao committed:
fix: use hasattr() to check if collate_fn is in dataloader

fix bug caused by PR #1596

TingquanGao committed:
fix: upgrade replace_sub() to upgrade_sublayer()

supplement to PR #1581

Commit:
Update quick_start_recognition.md

Commit:
Update quick_start_recognition.md

Commit:
add custom sampler;no eval; multicard eval

TingquanGao committed:
fix: fix min_lr & warmup_lr

JamesLim-sy committed:
first

Sibo2rr committed:
change subtitle order

Sibo2rr committed:
change MultiScaleSamplerDDP into MultiScaleSampler

Sibo2rr committed:
fix height error

TingquanGao committed:
doc: add the discription doc of theseus layer

littletomatodonkey committed:
add dist of rec model (#1574)

* add distillation loss func and rec distillation

JamesLim-sy committed:
drop useless settings

littletomatodonkey committed:
fox dist err (#1621)

* fox dist err

* fix init

* fix init

RainFrost1 committed:
fix en doc bugs for mainbody_detection_en vector_search_en classification_en recognition_en

TingquanGao committed:
feat: support specify return_stages to update_res()

RainFrost1 committed:
fix en doc bugs

TingquanGao committed:
doc: update wechat qr code

zhangbo9674 committed:
fix amp with distribute bug

zhangbo9674 committed:
refine code

zhangbo9674 committed:
fix eval bug

Commit:
update mosilian example

TingquanGao committed:
fix: change the returned result from dict to list

TingquanGao committed:
docs: supplement

TingquanGao committed:
fix: args error

zhangbo9674 committed:
use fp32 to eval

zhangbo9674 committed:
add save_dtype

weisy11 committed:
dbg

Commit:
add crop example

TingquanGao committed:
fix: fix bug

bug is that error rasied by calling update_res() when sublayer does not have _save_sub_res_hook() because nn.Sequential and nn.LayerList is not TheseusLayer.

TingquanGao committed:
docs: supplement

TingquanGao committed:
refactor: change params to be consistent with amp

TingquanGao committed:
fix: enable amp only in training

RainFrost1 committed:
fix en doc bugs

RainFrost1 committed:
fix en doc bugs

RainFrost1 committed:
fix alt_gvt_small_bug

TingquanGao committed:
docs: fix links, test=document_fix

TingquanGao committed:
docs: fix invalid links

TingquanGao committed:
doc: update wechat qr code, test=document_fix

TingquanGao committed:
fix

TingquanGao committed:
docs: add the description and link of vdl

TingquanGao committed:
fix: fix vdl makedir

TingquanGao committed:
refactor: raise warning when gpu numbers is not 4

TingquanGao committed:
fix: move class_num from Global to Arch

TingquanGao committed:
fix: raise warning when using Global.class_num

Commit:
support onnx inference  for cls; add readme

Commit:
update readme.md

Commit:
Update readme.md

Commit:
Update readme.md

Commit:
Update readme.md

Commit:
add onnx tipc for ResNet50_vd

Commit:
rm useless config

Commit:
fix predict_det.py error

Commit:
add insatll nvidia_dali in prepare.sh

TingquanGao committed:
docs: fix

TingquanGao committed:
docs: update wechat qr code

RainFrost1 committed:
add ppshitu lite

RainFrost1 committed:
fix ppshitu_lite bugs

Commit:
rename dali test configure

weisy11 committed:
add label_list file

weisy11 committed:
dbg

weisy11 committed:
dbg

RainFrost1 committed:
fix ppshitu_lite bugs and fix README.md

RainFrost1 committed:
add pp-shitu c++ link in readme

Commit:
modify serving bug;center image

Commit:
Update quick_start_recognition.md

Commit:
Update paddle_serving_deploy.md

Commit:
Update readme.md

RainFrost1 committed:
fix bugs and update readme for ppshitu_lite

RainFrost1 committed:
update readme for ppshitu_lite

RainFrost1 committed:
update readme for ppshitu_lite

Commit:
Update quick_start_recognition.md

Commit:
Update quick_start_recognition.md

RainFrost1 committed:
update readme  and demo result image for ppshitu_lite

TingquanGao committed:
docs: update wechat qr code, test=document_fix

TingquanGao committed:
docs: update wechat qr code, test=document_fix

TingquanGao committed:
docs: update wechat group qr code, test=document_fix

Commit:
fix serving doc problem

TingquanGao committed:
docs: update wechat group qr code, test=document_fix

Sibo2rr committed:
readthedoc in english

Sibo2rr committed:
fix: remove outdated files

weisy11 committed:
Update README_ch.md

weisy11 committed:
Update README_ch.md

cuicheng01 committed:
polish some docs

cuicheng01 committed:
update PP-LCNet_en.md

cuicheng01 committed:
update PP-LCNet_en.md

cuicheng01 committed:
update

weisy11 committed:
Update feature_extraction.md

weisy11 committed:
Update feature_extraction_en.md

Sibo2rr commented:
-
